pkg/credstash/backend.go

```diff
@@ -0,0 +1,172 @@
+// Package credstash implements backend for Credstash (that uses AWS KMS and DynamoDB)
+// Heavily inspired in github.com/ouzi-dev/credstash-operator using https://github.com/versent/unicreds
+package credstash
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/containersolutions/externalsecret-operator/pkg/backend"
+	"github.com/containersolutions/externalsecret-operator/pkg/utils"
+	unicreds "github.com/versent/unicreds"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+const (
+	defaultRegion          = "eu-west-2"
+	credstashVersionLength = 19
+)
+
+var (
+	log                     = ctrl.Log.WithName("credstash")
+	table                   = ""
+	configEncryptionContext = make(map[string]string)
```

Comment:
hey mate, are there any specific reasons for `table` and `configEncryptionContext` to be global variables? Would be possible to work with them inside the `SecretManagerClientStruct`?

Reply:
You are right, gonna move them ;)

Reply:
I gave this some thought, and I think it is simpler to leave them as global vars... If I add them to the Struct I would not be able to access them later in the code since there is no way to define fields like that for the Interface (and it is the interface that we actually define as the Client). I could create a Params Struct with Getter and Setters for SecretManagerClientStruct, but then this is very not idiomatic in go (for what I was able to check). I think it is way simpler to just leave it as global vars :smile:  Let me know what you think.

Reply:
Alright, go for it!